llvm/test/ObjectYAML/DXContainer/PSVv0-amplification.yaml

```diff
@@ -19,11 +19,11 @@ Parts:
       MaximumWaveLaneCount: 4294967295
       ResourceStride:       16
       Resources:
-        - Type:            1
+        - Type:            Sampler
           Space:           2
           LowerBound:      3
           UpperBound:      4
-        - Type:            128
+        - Type:            Invalid
```

Comment:
Sorry if this is a dumb question, but I'm not very familiar with the YAML printer. Is this printing the enum value of `0` (ie, "Invalid") or is the value in memory `128` and anything outside of the enum range prints "Invalid"? If it's `0`, what part of this change caused the value to change?

Reply:
My understanding is that for enum class types, you cannot say 128 anymore.
Have to choose from the enum names.

Reply:
I'm just confused because I see some new definitions and I see test changes, but I don't see the producer of the value changing anywhere so it isn't clear to me what the actual difference in behaviour is. I'm guessing the new `ScalarEnumerationTraits` specializations are the cause, but is the value different in the actual DXContainer or is it just a display thing with the YAML?

Reply:
The value should not change, unless it does not have a mapping to the enum values.
It just YAML now require the names instead of integer.

Reply:
Seems kind of unfortunate to not see the actual value that's in the structure in our debug tools - couldn't this mean that we end up putting garbage values in the actual container without noticing it in our tests? I guess if that's how the yaml printing works it's a wider issue and doesn't need to block this change in particular.

Reply:
Yes. `128` is garbage value which should not happen.